File: dataset_reduzido/avatar/Python/Code/codeforces_7_B.py
t , m = map ( int , input ( ).split ( ) )
alocuente = [ 0 ] * m
ind = 1
for i in range ( t ) :
    op = input ( ).split ( )
    if op [ 0 ] == 'alloc' :
        pos = 0
        for j in range ( m ) :
            if alocuente [ j ] == 0 :
                pos += 1
                if pos == int ( op [ 1 ] ) :
                    alocuente [ j - int ( op [ 1 ] ) + 1 : j + 1 ] = [ ind ] * int ( op [ 1 ] )
                    print ( ind )
                    ind += 1
                    break
            else :
                pos = 0
        else :
            print ( "NULL" )
    if op [ 0 ] == 'erase' :
        pos = 0
        if int ( op [ 1 ] ) not in alocuente or int ( op [ 1 ] ) == 0 :
            print ( 'ILLEGAL_ERASE_ARGUMENT' )
        else :
            for j in range ( m ) :
                if int ( op [ 1 ] ) > 0 and alocuente [ j ] == int ( op [ 1 ] ) :
                    alocuente [ j ] = 0
    if op [ 0 ] == "defragment" :
        cnt = 0
        cnt = alocuente.count ( 0 )
        alocuente = [ j for j in alocuente if j != 0 ]
        for j in range ( cnt ) :
            alocuente.append ( 0 )

File: output/Llama-2-100/avatar/Python/Java/codeforces_7_B.java
import java.util.Scanner;

public class codeforces_7_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        int m = sc.nextInt();
        int[] alocuente = new int[m];
        int ind = 1;
        for (int i = 0; i < t; i++) {
            String op = sc.nextLine();
            if (op.equals("alloc")) {
                int pos = 0;
                for (int j = 0; j < m; j++) {
                    if (alocuente[j] == 0) {
                        pos++;
                        if (pos == Integer.parseInt(op.split(" ")[1])) {
                            int start = j - Integer.parseInt(op.split(" ")[1]) + 1;
                            int end = j + 1;
                            alocuente[start:end] = new int[Integer.parseInt(op.split(" ")[1])];
                            for (int k = 0; k < Integer.parseInt(op.split(" ")[1]); k++) {
                                alocuente[start + k] = ind;
                                ind++;
                            }
                            System.out.println(ind);
                            break;
                        }
                    } else {
                        pos = 0;
                    }
                } else {
                    System.out.println("NULL");
                }
            } else if (op.equals("erase")) {
                int pos = 0;
                if (Integer.parseInt(op.split(" ")[1]) == 0 || !alocuente.contains(Integer.parseInt(op.split(" ")[1]))) {
                    System.out.println("ILLEGAL_ERASE_ARGUMENT");
                } else {
                    for (int j = 0; j < m; j++) {
                        if (alocuente[j] == Integer.parseInt(op.split(" ")[1]) && Integer.parseInt(op.split(" ")[1]) > 0) {
                            alocuente[j] = 0;
                        }
                    }
                }
            } else if (op.equals("defragment")) {
                int cnt = 0;
                cnt = alocuente.stream().filter(x -> x != 0).count();
                alocuente = Arrays.asList(alocuente).stream().filter(x -> x != 0).toArray(new int[cnt]);
                for (int j = 0; j < cnt; j++) {
                    alocuente.add(0);
                }
            }
        }
    }
}
